pkg/collect/redact.go

```diff
@@ -1,20 +1,101 @@
 package collect
 
 import (
+	"archive/tar"
+	"bytes"
+	"encoding/binary"
+	"io"
+	"path/filepath"
+
 	troubleshootv1beta1 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta1"
 	"github.com/replicatedhq/troubleshoot/pkg/redact"
 )
 
 func redactMap(input map[string][]byte, additionalRedactors []*troubleshootv1beta1.Redact) (map[string][]byte, error) {
 	result := make(map[string][]byte)
 	for k, v := range input {
-		if v != nil {
-			redacted, err := redact.Redact(v, k, additionalRedactors)
+		if v == nil {
+			continue
+		}
+		//If the file is a .tar file, it must not be redacted. Instead it is decompressed and each file inside the
+		//tar is decompressed, redacted and compressed back into the tar.
+		if filepath.Ext(k) == ".tar" {
+			tarFile := bytes.NewBuffer(v)
+			unRedacted, tarHeaders, err := untarFile(tarFile)
+			if err != nil {
+				return nil, err
+			}
+			redacted, err := redactMap(unRedacted, additionalRedactors)
+			if err != nil {
+				return nil, err
+			}
+			result[k], err = tarFiles(redacted, tarHeaders)
 			if err != nil {
 				return nil, err
 			}
-			result[k] = redacted
+			//Content of the tar file was redacted. Continue to next file.
+			continue
+		}
+		redacted, err := redact.Redact(v, k, additionalRedactors)
```

Comment:
This can corrupt `tgz` and `tar.gz`  files.

Reply:
Yes! Only when the file is inside the folder. I suggested a possible upgrade to the code to decompress .tgz and .tar.gz files too. If you think it would be useful I can work in this functionality!

Reply:
we should address this in a separate pr if we choose to